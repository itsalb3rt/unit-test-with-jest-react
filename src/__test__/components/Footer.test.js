import React from 'react';
import { mount } from 'enzyme';
import {create} from 'react-test-renderer';
import Footer from '../../components/Footer';

describe('<Footer />', ()=>{
    const footer = mount(<Footer />);    

    test('Render del component Footer', () => {
        expect(footer.length).toEqual(1); // check is component render
    });

    test('Render del titulo', () => {
        // Find a element by class name and check is text contain equeal to `any`
        expect(footer.find('.Footer-title').text()).toEqual("Platzi Store");
    });

});

describe('Footer Snapshot', () => {
    test('Comprobar la UI del component Footer', () => {
        const footer = create(<Footer />);
        /**
         * The first time you run the test the snapshot is created 
         * and when you run the test again just compare the snapshot with the 
         * component in these next tests
         */
        expect(footer.toJSON()).toMatchSnapshot();
    })
})