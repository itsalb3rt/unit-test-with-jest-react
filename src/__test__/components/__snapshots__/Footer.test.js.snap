// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer Snapshot Comprobar la UI del component Footer 1`] = `
<div
  className="Footer"
>
  <p
    className="Footer-title"
  >
    Platzi Store
  </p>
  <p
    className="Footer-copy"
  >
    Todos los Izquierdos Reservados
  </p>
</div>
`;
